### ISSUE TITLE
Unable to export Intune configuration

### ISSUE THREAD

**Issue by ronzipp** [microsoft/Microsoft365DSC#1934]
I'm not sure if this is one or two issues.  It fails to extract some of the components and it tells me that it cannot load Microsoft.Graph.Authentication.  I've tried both Credential and app/secret authentications but no luck.  Also, it appears the version 1.9.6.0 of Microsoft.Graph.Authentication is installed but when I try to uninstall, it says it cannot do it because of dependencies.  Any assistance would be greatly appreciated.

Export-M365DSCConfiguration -Components @("IntuneAppConfigurationPolicy", "IntuneAppProtectionPolicyAndroid", "IntuneAppProtectionPolicyiOS", "IntuneDeviceCategory", "IntuneDeviceCompliancePolicyAndroid", "IntuneDeviceCompliancePolicyAndroidDeviceOwner", "IntuneDeviceCompliancePolicyAndroidWorkProfile", "IntuneDeviceCompliancePolicyiOs", "IntuneDeviceCompliancePolicyMacOS", "IntuneDeviceCompliancePolicyWindows10", "IntuneDeviceConfigurationPolicyAndroidDeviceOwner", "IntuneDeviceConfigurationPolicyAndroidWorkProfile", "IntuneDeviceConfigurationPolicyiOS", "IntuneDeviceConfigurationPolicyWindows10", "IntuneDeviceEnrollmentLimitRestriction", "IntuneDeviceEnrollmentPlatformRestriction") -Credential $Credential
cmdlet Get-Credential at command pipeline position 1
Supply values for the following parameters:
Connecting to {MicrosoftGraph}...✅
[1/16] Extracting [IntuneAppConfigurationPolicy]...❌
[2/16] Extracting [IntuneAppProtectionPolicyAndroid]...❌
[3/16] Extracting [IntuneAppProtectionPolicyiOS]...❌
[4/16] Extracting [IntuneDeviceCategory]...❌
[5/16] Extracting [IntuneDeviceCompliancePolicyAndroid]...✅
[6/16] Extracting [IntuneDeviceCompliancePolicyAndroidDeviceOwner]...❌
[7/16] Extracting [IntuneDeviceCompliancePolicyAndroidWorkProfile]...✅
[8/16] Extracting [IntuneDeviceCompliancePolicyiOs]...✅
[9/16] Extracting [IntuneDeviceCompliancePolicyMacOS]...✅
[10/16] Extracting [IntuneDeviceCompliancePolicyWindows10]...✅
[11/16] Extracting [IntuneDeviceConfigurationPolicyAndroidDeviceOwner]...✅
[12/16] Extracting [IntuneDeviceConfigurationPolicyAndroidWorkProfile]...❌
Could not load file or assembly 'Microsoft.Graph.Authentication, Version=1.9.5.0, Culture=neutral, PublicKeyToken=31bf3856ad364e35' or one of its dependencies. The
 system cannot find the file specified.
[13/16] Extracting [IntuneDeviceConfigurationPolicyiOS]...❌
[14/16] Extracting [IntuneDeviceConfigurationPolicyWindows10]...❌
[15/16] Extracting [IntuneDeviceEnrollmentLimitRestriction]...❌
[16/16] Extracting [IntuneDeviceEnrollmentPlatformRestriction]...❌
⌛ Export took {30 seconds}

**Comment by ronzipp** [comment]
Thank you for the response.
When I ran 'Uninstall-M365DSCOutdatedDependencies' it gave an error that it could not uninstall 'Microsoft.Graph.Authentication 1.9.6.0'.  and that I had uninstall all the dependencies first.   Here are the dependencies I removed:
'Microsoft.Graph.Applications,Microsoft.Graph.DeviceManagement,Microsoft.Graph.DeviceManagement.Administration,Microsoft.Graph.DeviceManagement.Enrolment,Microsoft.Graph.Devices.CorporateManagement,Microsoft.Graph.Groups,Microsoft.Graph.Identity.DirectoryManagement,Microsoft.Graph.Identity.Governance,Microsoft.Graph.Identity.SignIns,Microsoft.Graph.Planner,Microsoft.Graph.Teams,Microsoft.Graph.Users'.
After they were removed I uninstalled 'Microsoft.Graph.Authentication 1.9.6.0'.
Next, I ran 'Update-M365DSCDependencies' which appears to have installed 'Microsoft.Graph.Authentication 1.9.6.0' again as I am still getting the exact same error as in my original post.

**Comment by andikrueger** [comment]
Could you try to uninstall the outdated dependencies again? 

1. Update-Module Microsoft365DSC
2. Update-M365DSCDependencies
3. Uninstall-M365DSCOutdatedDependencies

**Comment by martynstar** [comment]
I solved most of this by running it via Terminals app

**Comment by andikrueger** [comment]
Thanks for the udpate. I'll close this issue for now.

**Comment by OlivierWaterloos** [comment]
Hello,

I have this error since the version 1.21.xx. I have installed the latest one (1.22.629.1 Microsoft365DSC), I have Update-M365DSCDependencies and Uninstall-M365DSCOutdatedDependencies. 
Still the same.
I use this command:
Export-M365DSCConfiguration -Path $DSCPathWork -FileName $FilePSName  -ApplicationId $ApplicationId -CertificateThumbprint $CertificateThumbprint -TenantId $TenantId  -Components @("IntuneAppConfigurationPolicy") 

I have tried with -Credential $UserCredential --> same issue.
Other solution?
Thanks
